src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,22 +160,26 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
-                {
-                    if (AadIssuerV2 == null)
-                        AadIssuerV2 = CreateConfigManager(AadAuthorityV2).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
+                string AadIssuer;
+                BaseConfigurationManager configurationManager;
 
-                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
-                        return issuer;
-                }
+                if (validationParameters.ConfigurationManager != null)
+                    configurationManager = validationParameters.ConfigurationManager;
                 else
                 {
-                    if (AadIssuerV1 == null)
-                        AadIssuerV1 = CreateConfigManager(AadAuthorityV1).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV1, tenantId, issuer))
-                        return issuer;
+                    if (IsV2Authority)
+                        configurationManager = ConfigurationManagerV2;
+                    else
+                        configurationManager = ConfigurationManagerV1;
                 }
+
+                if (validationParameters.ValidateIssuerWithLKG)
+                    AadIssuer = configurationManager.LastKnownGoodConfiguration.Issuer;
```

Comment:
This LKG can be v1 or v2, it would fail if the LKG is v1 and current token is v2 and viceversa.

Reply:
it will check the token type and then decide to use configurationManagerv1 or configurationManagerv2